File: client/src/types/domain/donation.ts
export type DonationStatus = "pending" | "completed" | "failed" | "refunded";

export interface Donation {
  id: number;
  userId: number;
  displayName?: string;
  amount: number;
  currency: string;
  purpose: string;
  transactionId?: string;
  status: DonationStatus;
  receiptUrl?: string | null;
  createdAt: string;
}

export interface DonationStats {
  totalAmount: number | null;
  totalDonations: number;
}

export interface CreateDonationRequest {
  amount: number;
  currency?: string;
  purpose: string;
}

File: server/alumniweb/src/main/java/com/alumniweb/alumniweb/model/Donation.java
package com.alumniweb.alumniweb.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.SourceType;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
@Entity @Table(name = "donation")
public class Donation {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal amount;

    @Column(length = 3)
    @Builder.Default
    private String currency = "INR";

    @Column(length = 200)
    private String purpose;

    @Column(name = "transaction_id", length = 100)
    private String transactionId;

    @Column(nullable = false, length = 20)
    @Builder.Default
    private String status = "PENDING";

    @Column(name = "receipt_url", length = 500)
    private String receiptUrl;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @CreationTimestamp(source = SourceType.DB)
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp(source = SourceType.DB)
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}
